codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ServerDowngrader.java

```diff
@@ -78,9 +80,10 @@ protected void decode(ChannelHandlerContext ctx, Http2StreamFrame frame, List<Ob
                     out.add(full);
                 }
             } else {
-                out.add(HttpConversionUtil.toHttpRequest(id, headersFrame.headers(), validateHeaders));
+                HttpRequest req = HttpConversionUtil.toHttpRequest(id, headersFrame.headers(), validateHeaders);
+                HttpUtil.setTransferEncodingChunked(req, true);
```

Comment:
Oh, I see. Yeah, it looks like content-less requests should generally be OK. (I will note that there is some fuzziness whether they _must_ have EOS=true from an HTTP/2 standpoint. The spec [in an example](http://httpwg.org/specs/rfc7540.html#rfc.section.8.1.3) seems to encourage doing so, but I [don't see any explicit language](http://httpwg.org/specs/rfc7540.html#rfc.section.8.1) disallowing using a trailing DATA with EOS=true.)